atlasdb-config/src/main/java/com/palantir/atlasdb/factory/BroadcastDialogueServiceProvider.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import com.palantir.atlasdb.config.AuxiliaryRemotingParameters;
+import com.palantir.atlasdb.config.ImmutableAuxiliaryRemotingParameters;
+import com.palantir.atlasdb.config.ImmutableServerListConfig;
+import com.palantir.atlasdb.config.ServerListConfig;
+import com.palantir.atlasdb.http.AtlasDbHttpProtocolVersion;
+import com.palantir.atlasdb.http.AtlasDbRemotingConstants;
+import com.palantir.atlasdb.http.v2.DialogueClientOptions;
+import com.palantir.atlasdb.http.v2.DialogueShimFactory;
+import com.palantir.atlasdb.http.v2.ImmutableRemoteServiceConfiguration;
+import com.palantir.atlasdb.http.v2.RemoteServiceConfiguration;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.clients.DialogueClients;
+import com.palantir.refreshable.Refreshable;
+
+/**
+ * Provides a mechanism for creating individual proxies that use Dialogue for communication for each node in the
+ * cluster.
```

Comment:
It's worth calling out `AtlasDbDialogueServiceProvider` and explaining why this is different.

I'd even consider naming this `BroadcastDialogueClientFactory` or something along those lines, mainly because `Service` within the context of AtlasDB has a different meaning (which is generally the one in `AtlasDbDialogueServiceProvider`).